### PR TITLE
Add submit status for loan

### PR DESCRIPTION
**Pull request by dawid-pawulski**

### CODE REVIEW

File: src/HE.InvestmentLoans.BusinessLogic/LoanApplication/Repositories/LoanApplicationRepository.cs [hunk 1]
```diff
@@ -166,6 +166,17 @@ public async Task Save(LoanApplicationEntity loanApplication, CancellationToken
         LegacySave(loanApplication.LegacyModel);
     }
 
+    public void Submit(LoanApplicationViewModel loanApplication, CancellationToken cancellationToken)
+    {
+        var request = new invln_changeloanapplicationexternalstatusRequest
+        {
+            invln_loanapplicationid = loanApplication.ID.ToString(),
+            invln_statusexternal = 858110001, // that line will be changed
```
Comment (dawid-pawulski) [inline src/HE.InvestmentLoans.BusinessLogic/LoanApplication/Repositories/LoanApplicationRepository.cs:174]: That will be changed

File: src/HE.InvestmentLoans.BusinessLogic/LoanApplication/CommandHandlers/SubmitApplicationCommandHandler.cs [hunk 2]
```diff
@@ -19,5 +19,7 @@ public SubmitApplicationCommandHandler(ILoanApplicationRepository loanApplicatio
     public async Task Handle(SubmitApplicationCommand request, CancellationToken cancellationToken)
     {
         _loanApplicationRepository.Save(request.Model, await _loanUserContext.GetSelectedAccount());
+
+        _loanApplicationRepository.Submit(request.Model, cancellationToken);
```
Comment (jakub-lotko) [inline src/HE.InvestmentLoans.BusinessLogic/LoanApplication/CommandHandlers/SubmitApplicationCommandHandler.cs:23]: I think we should check if application can be submitted aka. all sections are completed. That method could be placed inside LoanApplicationEntity to follow OO approach.

File: src/HE.CRM.Common/DtoMapping/LoanApplicationDtoMapper.cs [hunk 3]
```diff
@@ -68,7 +73,7 @@ public static LoanApplicationDto MapLoanApplicationToDto(invln_Loanapplication l
             {
                 fundingReason = MapFundingReasonOptionSetToString(loanApplication.invln_FundingReason),
                 numberOfSites = loanApplication.invln_NumberofSites?.ToString(),
-                loanApplicationStatus = MapApplicationStatusFromDtoToRegular(loanApplication.invln_ExternalStatus),
+                //loanApplicationStatus = MapApplicationStatusFromDtoToRegular(loanApplication.invln_ExternalStatus),
```
Comment (lukasz-ciura-pwc) [inline src/HE.CRM.Common/DtoMapping/LoanApplicationDtoMapper.cs:76]: Remove line if code is not needed.

File: src/HE.InvestmentLoans.BusinessLogic/LoanApplication/Entities/LoanApplicationEntity.cs [hunk 4]
```diff
@@ -43,6 +49,40 @@ public void SetId(LoanApplicationId newId)
         SyncToLegacyModel();
     }
 
+    public void Submit(ICanSubmitLoanApplication canSubmitLoanApplication, CancellationToken cancellationToken)
+    {
+        if (!IsReadyToSubmit())
+        {
+            canSubmitLoanApplication.Submit(Id, cancellationToken);
```
Comment (lukasz-ciura-pwc) [inline src/HE.InvestmentLoans.BusinessLogic/LoanApplication/Entities/LoanApplicationEntity.cs:56]: `await` is missing

File: src/HE.InvestmentLoans.BusinessLogic/LoanApplication/Repositories/LoanApplicationRepository.cs [hunk 5]
```diff
@@ -166,6 +189,19 @@ public async Task Save(LoanApplicationEntity loanApplication, CancellationToken
...
+    {
+        var crmSubmitStatus = ApplicationStatusMapper.MapToCrmStatus(ApplicationStatus.Submitted);
+
+        var request = new invln_changeloanapplicationexternalstatusRequest
+        {
+            invln_loanapplicationid = loanApplicationId.ToString(),
+            invln_statusexternal = crmSubmitStatus,
+        };
+
+        _serviceClient.ExecuteAsync(request, cancellationToken);
```
Comment (lukasz-ciura-pwc) [inline src/HE.InvestmentLoans.BusinessLogic/LoanApplication/Repositories/LoanApplicationRepository.cs:202]: `await` is missing

**Review by jakub-lotko — APPROVED** [review]

**Pull request merged**